### DISCUSSION TITLE
How to restart an linked Actor inside the `on_link_died`

### DISCUSSION THREAD

**Discussion by jBernavaPrah** [tqwewe/kameo#93] [Q&A] [answered]
Hello!

I'm trying to understand the best approach for managing failing actors while attached to a manager. 

These actors are responsible for performing various utilities, but they could occasionally fail, and the manager needs to restart them. 

I looked into the PoolActor, and it suggests using a factory [see](https://github.com/tqwewe/kameo/blob/176ec7b5cd78cc3bc9c318a48a819438ca9b6641/src/actor/pool.rs#L143). 

Is this the only way available? 

Thanks a lot!

**Comment by tqwewe** [comment, marked as answer]
Hi @jBernavaPrah !

Restarting an actor is as simple as returning None in its `Actor::on_panic` implementation. By default, actors die when panic, but this behaviour can be overridden by returning None in this method instead
